app/javascript/controllers/tom_select_controller.js

```diff
@@ -7,7 +7,7 @@ export default class extends Controller {
         plugins: ['remove_button'],
         create: true,
         createOnBlur: true,
-        persist: false
+        persist: false,
```

Comment:
I found this issue https://github.com/orchidjs/tom-select/discussions/38 where the onItemAdd callback was suggested and seems to work for clearing the search text after adding a category
```suggestion
        persist: false,
        onItemAdd: function() {
          this.setTextboxValue('');
          this.refreshOptions();
        }
```